### PR TITLE
Receive push notifications for incoming calls

### LINKED ISSUE #369: Receive push notifications (android)
Push notifications will be used as a mean to notify users of an incoming call

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
This PR allows to inform the user of an incoming call through **Push notifications**. A click on the notification leads the user to the premium video call page.

## Implemented
- use `capacitor/core` - `PushNotifications` & `LocalNotifications`
- add `@capacitor-community/fcm` plugin to support subscribing to topics
- fix android build missing support libraries with `jetifier`
- create default notification channel (setting high priority) to display the notification when the app is in the foreground (otherwise only sound+status-bar-icon)
- on notification (push/local) click -> extract call information from the payload and lead the user to the premium call page
- add generic `Snackbar` component
- display notifications related error messages to the user with `Snackbar`
- set application icon as local notification status bar icon
- add API endpoint to test sending a topic notification (active only when using emulator)

## Further enhancements
- fix app not compiling on android v26 (don't know if regression or not)
- url translation for premium video call -> on app opened from notification, how to know when translations are ready ?
- sticky notifications simulating incoming call
- error joining call (`cannot read property 'addEventListener' of undefined - navigator.mediaDevices.addEventListener`) -> may need to use `CustomTabs`
- check push notifications [icons recommandations](https://capacitorjs.com/docs/apis/push-notifications#push-notifications-icon)
- fix when error message is hidden, success message flashes
- test the app is correctly receiving a notification when the app is closed

## Tests on emulator
1. app (background) -> receive push notif -> click on it -> open video call page
2. app (foreground) -> receive push notif -> manually schedule local notification -> open video call page
3. app (background, screen locked) -> receive push notif -> click on it -> open the video call page

## Screenshot
- foreground notification:
foreground-notification
- status bar notification icon:
app-background-notification

Fix #369

### CODE REVIEW

File: functions/src/api/index.ts [hunk 1]
```diff
@@ -24,6 +26,12 @@ app.use('/api/v1', routerV1)
 app.use('/api/v1-private', routerPrivate)
 app.use('/api/v1-docs', swaggerUi.serve, swaggerUi.setup(swaggerSpec))
 
+if (isUsingEmulator()) {
+    app.get('/api/send-topic-notification', (req, res) =>
```
Comment (HugoGresse) [inline functions/src/api/index.ts:30]: le truc salace :D 
je l'enlèverai quand je ferais l'api pour les notif

File: src/components/App/Snackbar/Snackbar.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    return (
+        <div id="snackbar" className={classes.root}>
+            <MaterialSnackbar
+                open={snackbar.isDisplayed}
+                autoHideDuration={3000}
+                onClose={handleClose}>
+                {snackbar.isError ? (
+                    <Alert onClose={handleClose} severity="error">
+                        error
```
Comment (HugoGresse) [inline src/components/App/Snackbar/Snackbar.tsx:43]: // TODO

File: src/components/App/Snackbar/Snackbar.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    </Alert>
+                ) : (
+                    <Alert onClose={handleClose} severity="success">
+                        success message hey
```
Comment (HugoGresse) [inline src/components/App/Snackbar/Snackbar.tsx:47]: // TODO

File: src/components/App/Snackbar/snackbarReducer.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import produce, { Draft } from 'immer'
+import {
+    HIDE_SNACKBAR,
+    SHOW_ERROR_SNACKBAR,
+    SHOW_SUCCESS_SNACKBAR,
+} from './snackbarActionTypes'
+import { SnackbarState } from './snackbarSelector'
+
+const initialState = {
+    snackbar: {
```
Comment (HugoGresse) [inline src/components/App/Snackbar/snackbarReducer.ts:10]: it's already the snackbar, no need to have another level there

File: src/services/local-notifications.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import {
+    LocalNotificationActionPerformed,
+    NotificationChannel,
+    Plugins,
+} from '@capacitor/core'
+const { LocalNotifications } = Plugins
+export class LocalNotificationsService {
+    static async schedule(title?: string, body?: string, extra?: any) {
+        const ONE_SECOND_FROM_NOW = new Date(Date.now() + 1000)
```
Comment (HugoGresse) [inline src/services/local-notifications.ts:9]: lol

**Review by HugoGresse — COMMENTED** [review]
gj

File: functions/src/api/index.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply functions/src/api/index.ts:30]: why ? It allows to have an easy way to test notifications without bothering production code

File: src/components/App/Snackbar/snackbarReducer.ts (hunk 4 above)
Comment (MattiaPrimavera) [reply src/components/App/Snackbar/snackbarReducer.ts:10]: I based this on @ntchou work on users

File: src/services/local-notifications.ts (hunk 5 above)
Comment (MattiaPrimavera) [reply src/services/local-notifications.ts:9]: clean code ++

File: src/components/App/Snackbar/snackbarReducer.ts (hunk 4 above)
Comment (HugoGresse) [reply src/components/App/Snackbar/snackbarReducer.ts:10]: hmm, does not seems to be a good idea either. 
so here the state is actually `state.snackbar.snackbar` which make it redondent.

File: functions/src/api/index.ts (hunk 1 above)
Comment (HugoGresse) [reply functions/src/api/index.ts:30]: oué, je changerai plus tard :p

**Review by HugoGresse — APPROVED** [review]

**Pull request merged**